### PR TITLE
feat: add get messages usecase from kalium (AR-970)

### PR DESCRIPTION
**Pull request by borichellow**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-970" title="AR-970" target="_blank">SubtaskAR-970</a>  Add a UseCase for getting most recent messages from a conversation
  </td></table>
  <br />
 

[Task-970](https://wearezeta.atlassian.net/browse/AR-970)

# What's new in this PR?

Displaying received messages in real-time. 

ezgif com-gif-maker (2)

### CODE REVIEW

File: app/src/main/kotlin/com/wire/android/ui/home/HomeViewModel.kt [hunk 1]
```diff
@@ -2,18 +2,23 @@ package com.wire.android.ui.home
...
 import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @ExperimentalMaterial3Api
 @HiltViewModel
 class HomeViewModel
 @Inject constructor(
     private val navigationManager: NavigationManager,
+    private val listenToEventsUseCase: ListenToEventsUseCase,
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/ui/home/HomeViewModel.kt:21]:
```suggestion
    private val listenToEvents: ListenToEventsUseCase,
```

File: app/src/main/kotlin/com/wire/android/ui/home/HomeViewModel.kt [hunk 2]
```diff
@@ -35,4 +40,11 @@ class HomeViewModel
     }
 
     suspend fun navigateToUserProfile() = navigateTo(NavigationItem.UserProfile)
+
+    init {
+        //listen for the WebSockets updates and update DB accordingly
+        viewModelScope.launch {
+            listenToEventsUseCase.invoke()
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/ui/home/HomeViewModel.kt:47]:
```suggestion
            listenToEvents()
```

**Review by vitorhugods — CHANGES_REQUESTED** [review]
Small sugar syntax suggestions and questioning the need of declaring `Gson` dependency.

Other than that: 🚀 Let's see it fly!

File: app/src/main/kotlin/com/wire/android/navigation/NavigationItem.kt [hunk 3]
```diff
@@ -63,18 +63,18 @@ sealed class NavigationItem(
...
-        content = {
-            ConversationScreen(hiltViewModel())
-        },
+        content = { ConversationScreen(hiltViewModel()) },
         arguments = listOf(
             navArgument(CONVERSATION_ID_ARGUMENT) { type = NavType.StringType }
         )
     ) {
-        fun createRoute(conversationId: ConversationId) = "conversation/${conversationId}"
+        fun createRoute(conversationId: ConversationId) = "conversation/${conversationId.domain}@${conversationId.value}"
```
Comment (ohassine) [inline app/src/main/kotlin/com/wire/android/navigation/NavigationItem.kt:71]: We may need this somewhere else. How about an extension  function that do this concatenation ?

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt [hunk 4]
```diff
@@ -1,34 +1,61 @@
...
-        get() = _state
+    var conversationViewState by mutableStateOf(ConversationViewState())
+        private set
+
+    val conversationId: ConversationId? = savedStateHandle
+        .getLiveData<String>(NavigationItem.CONVERSATION_ID_ARGUMENT)
+        .value
+        ?.let {
+            val components = it.split("@")
+            ConversationId(components.last(), components.first())
```
Comment (ohassine) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt:42]: same here, an extension  function for this ?

**Review by ohassine — APPROVED** [review]

**Review by vitorhugods — APPROVED** [review]
🚀

**Pull request merged**